src/core/reaction_ensemble.cpp

```diff
@@ -0,0 +1,2061 @@
+	bool particle_inserted_too_close_to_another_one=true;
+	int max_insert_tries=1000;
+	int insert_tries=0;
+	double min_dist=current_reaction_system.exclusion_radius; //setting of a minimal distance is allowed to avoid overlapping configurations if there is a repulsive potential. States with very high energies have a probability of almost zero and therefore do not contribute to ensemble averages.
+	if(min_dist!=0){
+		while(particle_inserted_too_close_to_another_one && insert_tries<max_insert_tries) {
+			get_random_position_in_box(pos_vec);
+			place_particle(p_id,pos_vec);
+			//set type
+			set_particle_type(p_id, desired_type);
+			#ifdef ELECTROSTATICS
+			//set charge
+			set_particle_q(p_id, charge);
+			#endif
+			//set velocities
+			set_particle_v(p_id,vel);
+			double d_min=distto(pos_vec,p_id); //TODO also catch constraints with an IFDEF CONSTRAINTS here, but only interesting, when doing MD/ HMC because then the system might explode easily here due to high forces
+			insert_tries+=1;
+			if(d_min>current_reaction_system.exclusion_radius)
+				particle_inserted_too_close_to_another_one=false;
+		}
+	}else{
+		get_random_position_in_box(pos_vec);
+		place_particle(p_id,pos_vec);
+		//set type
+		set_particle_type(p_id, desired_type);	
+		//set velocities
+		set_particle_v(p_id,vel);
+		#ifdef ELECTROSTATICS
+		//set charge
+		set_particle_q(p_id, charge);
+		#endif
+	}
+	if(insert_tries>max_insert_tries){
+		printf("Error: Particle not inserted\n");
+		return -1;	
+	}
+	return p_id;
+}
+
+//the following 3 functions are directly taken from ABHmath.tcl
+double vec_len(double* vec){
+	double veclen=0;
+	for(int i=0;i<3;i++){
+		veclen+=pow(vec[i],2);	
+	}
+	return sqrt(veclen);
+}
+
+int vecnorm(double* vec, double desired_length){
+	double veclen=vec_len(vec);
+	for(int i=0;i<3;i++){
+		vec[i]=vec[i]/veclen*desired_length;	
+	}
+	return 0;
+}
+
+int vec_random(double* vecrandom, double desired_length){
+	//returns a random vector of length len
+	//(uniform distribution on a sphere)
+	//This is done by chosing 3 uniformly distributed random numbers [-1,1]
+	//If the length of the resulting vector is <= 1.0 the vector is taken and normalized
+	//to the desired length, otherwise the procedure is repeated until succes.
+	//On average the procedure needs 5.739 random numbers per vector.
+	//(This is probably not the most efficient way, but it works!)
+	//Ask your favorit mathematician for a proof!
+	while(1){
+		for(int i=0;i<3;i++){
+			vecrandom[i]=2*d_random()-1.0;
+		}
+		if (vec_len(vecrandom)<=1)
+			break;
+	}
+	vecnorm(vecrandom,desired_length);
+	return 0;
+}
+
+//for sorting integer lists, derived from http://rosettacode.org/wiki/Sort_an_integer_array#C, function is licensed under GPL
+int intcmp(const void *aa, const void *bb){ 
+    const int *a = (int*) aa, *b =(int*)  bb;
+    return (*a > *b) ? -1 : (*a < *b);
+}
+
+
+bool is_in_list(int value, int* list, int len_list){
+	if(len_list==0){
+		return true;
```

Comment:
removed the check for an empty list from this function and put it into the other places where it was needed